cou/apps/auxiliary_subordinate.py

```diff
@@ -35,3 +40,56 @@ def current_os_release(self) -> OpenStackRelease:
         :rtype: OpenStackRelease
         """
         return self.channel_codename
+
+
+@AppFactory.register_application(["ovn-central", "ovn-dedicated-chassis"])
+class OvnPrincipalApplication(OpenStackAuxiliaryApplication):
+    """Ovn principal application class."""
+
+    def pre_upgrade_plan(self, target: OpenStackRelease) -> list[Optional[UpgradeStep]]:
+        """Pre Upgrade planning.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :return: Plan that will add pre upgrade as sub steps.
+        :rtype: list[Optional[UpgradeStep]]
+        """
+        for unit in self.units:
+            validate_ovn_support(unit.workload_version)
+        return super().pre_upgrade_plan(target)
```

Comment:
This is a principal charm inherited from `OpenStackAuxiliaryApplication`, so it should be in `auxiliary.py`.